### DISCUSSION TITLE
How to unit test data validation at rest level

### DISCUSSION THREAD

**Discussion by tokidoki11** [loopbackio/loopback-next#10311] [Q&A]
https://loopback.io/doc/en/lb4/Validation-REST-layer.html

As you may already know loopback 4 embed ajv validation for schema on rest layer

I am wondering how to create unit test so i can ensure the validity of my settings

https://loopback.io/doc/en/lb4/Testing-your-application.html#unit-test-your-controllers

i think the request  model validation unit test should be run on controller unit 
But i cant find any example of doing so

**Comment by yiev** [comment]
I think acceptance tests are the way for testing the schema validation.
https://loopback.io/doc/en/lb4/Testing-your-application.html#acceptance-end-to-end-testing

Here are some examples testing schema validation by expecting 422 (Unprocessable Entity, so most probably schema validation failure) as response code:
https://github.com/loopbackio/loopback4-example-shopping/blob/master/packages/shopping/src/__tests__/acceptance/user-management.controller.acceptance.ts
